### PR TITLE
Verbatim string interpolation guidance

### LINKED ISSUE #29765: Add an example for `@$` and formatting
See [Visual Studio feedback item](https://devdiv.visualstudio.com/DevDiv/_workitems/edit/1487580)

Our samples don't include a good example for an interpolated verbatim string.

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: c7aa6134-64e2-28a7-f7d0-d0c1512f4520
* Version Independent ID: 3317fd4a-9cd3-57cd-66f7-abb271e302e8
* Content: [Strings - C# Programming Guide](https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/strings/)
* Content Source: [docs/csharp/programming-guide/strings/index.md](https://github.com/dotnet/docs/blob/main/docs/csharp/programming-guide/strings/index.md)
* Product: **dotnet-csharp**
* Technology: **csharp-fundamentals**
* GitHub Login: @BillWagner
* Microsoft Alias: **wiwagn**

### PR DESCRIPTION
**Pull request by SeanKilleen**
## Summary

* [x] Adds guidance text to string guide
* [x] Adds code snippet showcasing verbatim string interpolation.

Fixes #29765.

### CODE REVIEW

File: docs/csharp/programming-guide/strings/index.md [hunk 1]
```diff
@@ -117,6 +117,14 @@ Beginning with C# 11, you can combine *raw string literals* with string interpol
 
 :::code language="csharp" source="./snippets/StringInterpolation.cs" id="InterpolationExample":::
 
+### Verbatim String Interpolation
+
+C# also allows verbatim string interpolation, for example across multiple lines, using the `$@` or `@$` syntax.
+
+To interpret escape sequences literally, use a [verbatim](../../language-reference/tokens/verbatim.md) string literal. An interpolated verbatim string starts with the `$` character followed by the `@` character. Starting with C# 8.0, you can use the `$` and `@` tokens in any order: both `$@"..."` and `@$"..."` are valid interpolated verbatim strings.
+
+:::code language="csharp" source="./snippets/VerbatimStringInterpolation.cs" id="VerbatimInterpolationExample":::
```
Comment (BillWagner) [inline docs/csharp/programming-guide/strings/index.md:126]: The tag in the code is "VerbatimStringInterpolation":

```suggestion
:::code language="csharp" source="./snippets/VerbatimStringInterpolation.cs" id="VerbatimStringInterpolation":::
```

**Comment by SeanKilleen** [comment]
OK, I think this is all good now. 👍 Ready for official review.

File: docs/csharp/programming-guide/strings/snippets/VerbatimStringInterpolation.cs [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+public static class VerbatimInterpolation
+{
+    public static void VerbatimInterpolationExample()
+    {
+        // <VerbatimStringInterpolation>
+        var jh = (firstName: "Jupiter", lastName: "Hammon", born: 1711, published: 1761);
+        Console.WriteLine($@"{jh.firstName} {jh.lastName}
+            was an African American poet born in {jh.born}.");
+        Console.WriteLine(@$"He was first published in {jh.published}
+        at the age of {jh.published - jh.born}.");
```
Comment (BillWagner) [inline docs/csharp/programming-guide/strings/snippets/VerbatimStringInterpolation.cs:13]: Another alternative is to outdent the 2nd lines:

```suggestion
        Console.WriteLine($@"{jh.firstName} {jh.lastName}
was an African American poet born in {jh.born}.");
        Console.WriteLine(@$"He was first published in {jh.published}
at the age of {jh.published - jh.born}.");
```

**Review by BillWagner — CHANGES_REQUESTED** [review]
Hi @SeanKilleen 

I like what you've added.

Right now, there is a mis-match between the code text and the output. That's because verbatim strings include all whitespace. This is important because we're adding *raw string literals* in C# 11. I've put in two different suggestions. Either one of them would work. I have a mild preference for changing the displayed output, but either is fine.

**Comment by SeanKilleen** [comment]
Thanks @BillWagner! I agree with your suggestion to change the output, and I committed it. 👍

**Review by BillWagner — APPROVED** [review]

**Pull request merged**